panoramapublic/src/org/labkey/panoramapublic/view/publish/pxValidationStatus.jsp

```diff
@@ -0,0 +1,852 @@
+        });
+    }
+
+    // -----------------------------------------------------------
+    // Displays the main validation summary panel
+    // -----------------------------------------------------------
+    function validationInfo(json) {
+
+        function getStatusCls(statusId) {
+            return statusId === 3 ? 'pxv-valid' : (statusId !== -1 ? 'pxv-invalid' : '');
+        }
+
+        function getMissingMetadataFields(missingFields) {
+            var list = '<ul>';
+            for (var i = 0; i < missingFields.length; i++) {
+                list += '<li>' + htmlEncode(missingFields[i]) + '</li>';
+            }
+            list += '</ul>';
+            return list;
+        }
+
+        function problemSummary(json) {
+            var problems = '';
+            if (json["missingMetadata"]) {
+                var updateMedataLink = LABKEY.ActionURL.buildURL('panoramapublic', 'showUpdateExperimentAnnotations', LABKEY.ActionURL.getContainer(), {id: <%=experimentAnnotationsId%>});
+                problems += '<li>Missing metadata: [' + link("Update Metadata", updateMedataLink, 'pxv-bold') + ']' + getMissingMetadataFields(json["missingMetadata"]) + '</li>';
+            }
+            if (json["modificationsValid"] === false) problems += '<li>Modifications without a Unimod ID</li>';
+            if (json["sampleFilesValid"] === false) problems += '<li>Missing raw data files</li>';
+            if (json["specLibsComplete"] === false) problems += '<li>Incomplete spectral library information</li>';
+            return '</br>Problems found: <ul>' + problems + '</ul>';
+        }
+
+        function getStatusValidHtml() {
+            return 'The data is valid for a "complete" ProteomeXchange submission.  ' +
+                    'You can view the validation details below.';
+        }
+
+        function getIncompleteDataHtml(json) {
+            return 'The data can be assigned a ProteomeXchange ID but it is not valid for a "complete" ProteomeXchange submission. ' +
+                    problemSummary(json) +
+                    'You can view the validation details in the tables below. ' +
+                    'For a "complete" submission try submitting after fixing the problems reported. ' +
+                    'Otherwise, you can continue with an incomplete submission.';
+        }
+
+        function getStatusInvalidHtml(json) {
+            return 'The data cannot be assigned a ProteomeXchange ID. ' +
+                    problemSummary(json) +
+                    'You can view the validation details in the tables below. ' +
+                    'Try submitting the data after fixing the problems reported. ' +
+                    'Otherwise, you can submit the data without a ProteomeXchange ID.';
+        }
+
+        function getStatusDetails(statusId, json) {
+            var html =  statusId === 3 ? getStatusValidHtml(json)
```

Comment:
Is it worth pulling these values into constants, even if they're just within this JSP?